AI/src/main/java/dk/gruppeseks/bodtrd/ai/AIProvider.java

```diff
@@ -20,57 +22,48 @@
 public class AIProvider implements AISPI
 {
     private boolean[][] _grid;
+    private boolean[][] _explored;
 
     @Override
-    public Position[] getPath(Position start, Position goal, World world)
+    public Stack<Position> getPath(Position start, Position goal, World world) throws NoPathException
     {
-        //The following is dummy-code, untill the map is fixed
-        boolean[][] testGrid =
-        {
-            {
-                true, true, true, true
-            },
-            {
-                true, false, true, false
-            },
-            {
-                true, false, true, true
-            },
-            {
-                true, true, false, true
-            }
-        };
-        //End of dummy-code
-
         _grid = world.getMap().getGrid();
+        _explored = new boolean[_grid.length][_grid[0].length];
```

Comment:
if we are going to initialize a new array for each call to getPath, then it makes no sense declaring it as a field